Assignments/Program.cs

```diff
@@ -1 +1,47 @@
-﻿Console.WriteLine("Hello, World!");
+﻿// ---------------------------------------------------------------------------------------
+// T5 - Prime Number Checker
+// ---------------------------------------------------------------------------------------
+using System;
+using System.Numerics;
+using static System.ConsoleColor;
+
+namespace Spark;
+
+public class T5 {
+   public static void Main () {
+      Console.WriteLine ("\x1B[4m" + "Prime Number Checker:-" + "\x1B[0m");
+      int count = 1;
+      string output;
+      int[] test = new int[] { 2, 3, 5, 11, 51, 61, 297, 313, 1381, 1663, 5157, 5153, 7871, 7873, 7919, 110503, 1257787, 3021379, 37156667 };
+      foreach (int i in test) {
+         Console.Write ($"{count++}. {i} ");
+         (output, Console.ForegroundColor) = IsPrime (i) ? ("Prime", Green) : ("Not a Prime", Red);
+         Console.WriteLine ($" - {output}");
+         Console.ResetColor ();
+      }
+      Console.Write ("\nNow, lets enter a number to check: ");
+      int s = Convert.ToInt32 (Console.ReadLine ());
+      (output, Console.ForegroundColor) = IsPrime (s) ? ("Prime", Green) : ("Not a Prime", Red);
+      Console.WriteLine ($"{output}");
+      Console.ResetColor ();
+   }
+
+   /// <summary>Returns true if the input number is a prime, else false</summary>
+   public static bool IsPrime (long num) {

```

Comment:
IsPrime has as input arg 'long' but it is always called with argument 'int'

Reply:
I presumed the user may prove a large value, so tried to pass a long instead of int.